docs/api-change-runbook.md

```diff
@@ -0,0 +1,264 @@
+
+## How to deprecate
+
+As outlined in the diagram above, sometimes we have to deprecate endpoints
+partially or entirely. In this section we will go through different situations
+where we have to deprecate something and ways of avoiding common pitfalls when
+doing so.
+
+### Deprecating endpoints
+
+Some paths in the flowchart above lead to deprecation. Based on the initial
+conditions, there are 2 major cases where we need to deprecate an endpoint:
+
+* *Changing an existing endpoint*
+  * Often happens because directly changing the endpoint would be a breaking
+    change.
+  * We usually create a clone of the old endpoint we want to deprecate and
+    make the necessary changes to it.
+  * We usually expose both endpoints in the next minor version while marking
+    the old one as deprecated.
+  * The old endpoint retains its previous name. When naming the new endpoint:
+    * for HTTP endpoints we follow a “per-endpoint versioning” scheme; in
+        cases where we can’t find a fitting name for the new endpoint, the
+        simplest way forward is to take the old URI and append `/v2` to it.
+    * for command line endpoints, we can usually find a different name for
+        the new endpoint.
+* *Deprecating an endpoint without adding a replacement to it*
+  * Often happens when we want to phase out a certain feature or
+    functionality, but doing so immediately would be a breaking change.
+  * We just mark the endpoint as deprecated.
+
+### Keeping Swagger updated
+
+Make sure that any changes you make in the code are also reflected in the
+swagger specification.
+
+Some tips:
+
+* There is nothing in the swagger file that shows whether an endpoint is
+  mandatory or optional, it’s all code logic.
+* Mandatory fields in a request or response body are marked with
+  `required: true` in the swagger definition. All other fields are optional.
+* If you need to redirect an endpoint, you have to clone the old one under the
+  new URI in the swagger specification.
+
+### Marking endpoints as deprecated
+
+When marking:
+
+* an HTTP endpoint as deprecated:
+  * Add a comment for the parsing function of the endpoint stating that it
+    is deprecated.
+  * Log a `warn!` message stating that the user accessed a deprecated
+    endpoint.
+  * Increment the `deprecatedHttpApi` metric.
+  * Include the `Deprecated` header in the response.
+* a header field in an HTTP endpoint as deprecated:
+  * Add a comment in the parsing function where we check the presence of the
+    header stating that it is deprecated.
+  * If the header is present, log a `warn!` message stating that the user
+    used a deprecated field.
+  * Increment the `deprecatedHttpApi` metric.
+  * Include the Deprecated header in the response.
+* a command line parameter as deprecated:
+  * Mention it is deprecated in the help message of the parameter in the
+    argument parser.
+  * Add it in the `warn_deprecated_parameters` function where we log it
```

Comment:
What is exactly is the use case for having such a metric?

Reply:
This is for all customers which want to track their API usage and check if they are ever using deprecated endpoints.